instrumentation/dropwizard/dropwizard-testing/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizard/DropwizardAsyncTest.java

```diff
@@ -0,0 +1,184 @@
+    return AsyncTestApp.class;
+  }
+
+  @Override
+  Class<?> testResource() {
+    return AsyncServiceResource.class;
+  }
+
+  @Override
+  protected void configure(HttpServerTestOptions options) {
+    super.configure(options);
+    // server spans are ended inside the JAX-RS controller spans
+    options.setVerifyServerSpanEndTime(false);
+  }
+
+  public static class AsyncTestApp extends Application<Configuration> {
+    @Override
+    public void initialize(Bootstrap<Configuration> bootstrap) {}
+
+    @Override
+    public void run(Configuration configuration, Environment environment) {
+      environment.jersey().register(AsyncServiceResource.class);
+    }
+  }
+
+  @Path("/")
+  public static class AsyncServiceResource {
+    final ExecutorService executor = Executors.newSingleThreadExecutor();
+
+    @GET
+    @Path("success")
+    public void success(@Suspended AsyncResponse asyncResponse) {
+      executor.execute(
+          () ->
+              controller(
+                  SUCCESS,
+                  () ->
+                      asyncResponse.resume(
+                          Response.status(SUCCESS.getStatus()).entity(SUCCESS.getBody()).build())));
+    }
+
+    @GET
+    @Path("query")
+    public void queryParam(
+        @QueryParam("some") String param, @Suspended AsyncResponse asyncResponse) {
+      executor.execute(
+          () ->
+              controller(
+                  QUERY_PARAM,
+                  () ->
+                      asyncResponse.resume(
+                          Response.status(QUERY_PARAM.getStatus())
+                              .entity("some=" + param)
+                              .build())));
+    }
+
+    @GET
+    @Path("redirect")
+    public void redirect(@Suspended AsyncResponse asyncResponse) {
+      executor.execute(
+          () -> {
+            try {
+              controller(
+                  REDIRECT,
+                  () ->
+                      asyncResponse.resume(
+                          Response.status(REDIRECT.getStatus())
+                              .location(new URI(REDIRECT.getBody()))
+                              .build()));
+            } catch (URISyntaxException e) {
+              throw new RuntimeException(e);
+            }
+          });
+    }
+
+    @GET
+    @Path("error-status")
+    public void error(@Suspended AsyncResponse asyncResponse) {
+      executor.execute(
+          () ->
+              controller(
+                  ERROR,
+                  () ->
+                      asyncResponse.resume(
+                          Response.status(ERROR.getStatus()).entity(ERROR.getBody()).build())));
+    }
+
+    @GET
+    @Path("exception")
+    public void exception(@Suspended AsyncResponse asyncResponse) {
+      executor.execute(
+          () -> {
+            try {
+              controller(
+                  EXCEPTION,
+                  () -> {
+                    Exception ex = new Exception(EXCEPTION.getBody());
```

Comment:
Could `options.setExpectedException(new IllegalStateException(EXCEPTION.getBody()));` and then use `IllegalStateException` here. Probably should change the default exception to `IllegalStateException` as the checked exception is annoying in java.

Reply:
> Probably should change the default exception to IllegalStateException as the checked exception is annoying in java.

is https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12555 the general idea of what you were thinking?